src/server/snap.rs

```diff
@@ -191,10 +208,30 @@ pub fn send_snap(
     let (sink, receiver) = client.snapshot()?;
 
     let send_task = async move {
-        let mut sink = sink.sink_map_err(Error::from);
-        sink.send_all(&mut chunks).await?;
-        sink.close().await?;
-        let recv_result = receiver.map_err(Error::from).await;
+        let send_and_recv = async {
+            let mut sink = sink.sink_map_err(Error::from);
+
+            #[cfg(feature = "failpoints")]
```

Comment:
sink.sink_map_err() and receiver.map_err(Error::from) should return error if there is a network error. Right ?

I am trying to understand that how the timeout is helping here other than it fails early and can retry.

Reply:
Yes, if the send fails due to network error, send snapshot will fail immediately, Timeout has no effect in this case. This timeout is to handle cases that the task does not fail but also not progress, thus if it can finished in specifical duration, that means there must be something wrong with the netwrok, we may just abort this task and try again latter.

Reply:
I am not sure if it is right RCA. Node was stuck in Snaphost pending state for days in our cluster. Are we saying that sink or receiver is hung internally forever.

Reply:
Sorry, I'm also not sure why the snapshot was stuck for such a long time. If you meet this problem again, we can catch a stack trace and it should help us locate the root cause of this stuck.